src/queue.rs

```diff
@@ -145,18 +145,13 @@ impl<M: GuestAddressSpace> DescriptorChain<M> {
         }
 
         let desc_size = size_of::<Descriptor>();
-        let desc_head = match mem.checked_offset(desc_table, (index as usize) * desc_size) {
+        let desc_head = match desc_table.checked_add((index as u64) * (desc_size as u64)) {
```

Comment:
Whenever the code is introduced earlier in the series, can you squash it there?

Reply:
fixed